### DISCUSSION TITLE
Export folder structure for backup purposes

### DISCUSSION THREAD

**Discussion by VadarVillage** [ciur/papermerge#547] [Q&A] [answered]
First off, I love this app. I have made a few Django apps for my own use that are far less ambitions than this app, you have clearly put a lot of work into this.

Do you have a function to export the directory structure and file names as they appear in the web UI?

I need to export the directory structure and files, then store an archive so that it can be used in windows file explorer. I reviewed the node.py model file, and I can make a script to do this on my own if I must. Your code is clean and easy to understand, thank you for making this tool.

**Comment by ciur** [comment, marked as answer]
I assume we are discussing here version 2.1.x.

> Do you have a function to export the directory structure and file names as they appear in the web UI?

yes, there are actually two django commands:
- [backup](https://github.com/papermerge/papermerge-core/blob/stable/2.1.x/papermerge/core/management/commands/backup.py)
- [restore](https://github.com/papermerge/papermerge-core/blob/stable/2.1.x/papermerge/core/management/commands/restore.py)

There is also a documentation page about them: https://docs.papermerge.io/backup_restore.html

I guess you are mostly interested in this module: https://github.com/papermerge/papermerge-core/blob/stable/2.1.x/papermerge/core/backup_restore/backup.py

If you will perform backup outside docker container, in other words on the file system which support symlinks, the backup
archive will include **the same structure as in web UI**, but instead of documents there are symlinks pointing to the
actual document files in media folder. Give it a try and you will understand what I mean (just keep in mind that your file system need to support symbolic links).

**Reply by VadarVillage** [reply]
Thank you! I appreciate you taking the time to make it easy for me. I'll check out the docs and that module.

I didn't want to start scanning all these documents before I knew what I was getting into.
